src/main/java/org/trigon/hardware/phoenix6/talonfx/TalonFXMotor.java

```diff
@@ -29,68 +48,147 @@ public TalonFXMotor(int id, String motorName, String canbus) {
     }
 
+    /**
+     * Sets the physics simulation of the motor. Needed for the motor to be used in simulation with accurate physics.
+     *
+     * @param physicsSimulation the simulation
+     */
     public void setPhysicsSimulation(MotorPhysicsSimulation physicsSimulation) {
         motorIO.setPhysicsSimulation(physicsSimulation);
     }
 
+    /**
+     * Applies both the real and simulation configurations to the motor.
+     * Having two different configurations allows for tuning motor behavior in simulation which might not perfectly mimic real life performance.
+     *
+     * @param realConfiguration       configuration to be used in real life
+     * @param simulationConfiguration configuration to be used in simulation
+     */
     public void applyConfigurations(TalonFXConfiguration realConfiguration, TalonFXConfiguration simulationConfiguration) {
         if (RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(simulationConfiguration);
         else
             motorIO.applyConfiguration(realConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used both in real life and in simulation.
+     *
+     * @param simulationAndRealConfiguration the configuration
+     */
     public void applyConfiguration(TalonFXConfiguration simulationAndRealConfiguration) {
         motorIO.applyConfiguration(simulationAndRealConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used when {@link RobotHardwareStats#isSimulation()} is false.
+     * Having two different configurations allows for tuning motor behavior in simulation which might not perfectly mimic real life performance.
+     *
+     * @param realConfiguration the configuration
+     */
     public void applyRealConfiguration(TalonFXConfiguration realConfiguration) {
         if (!RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(realConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used in simulation.
+     * Having two different configurations allows for tuning motor behavior in simulation which might not perfectly mimic real life performance.
+     *
+     * @param simulationConfiguration the configuration
+     */
     public void applySimulationConfiguration(TalonFXConfiguration simulationConfiguration) {
         if (RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(simulationConfiguration);
     }
 
+    /**
+     * Stops the motor.
+     */
     public void stopMotor() {
         motorIO.stopMotor();
     }
 
+    /**
+     * Gets a signal from the motor.
+     *
+     * @param signal the type of signal to get
+     * @return the signal
+     */
     public double getSignal(TalonFXSignal signal) {
         return motorInputs.getSignal(signal.name);
     }
 
+    /**
+     * Gets a threaded signal from the motor.
+     * Threaded signals use threading to process certain signals separately, keeping them running smoothly in the background.
```

Comment:
this is not the main reason for threaded signals, talk about high freq with the example of odom